src/transforms/parameterTransforms.ts

```diff
@@ -104,7 +105,7 @@ export function transformParameters(
   options: ClientOptions
 ): ParameterDetails[] {
   let parameters: ParameterDetails[] = [];
-  const { addCredentials } = getAutorestOptions();
+  const { addCredentials } = getSecurityInfoFromModel(codeModel.security);
```

Comment:
same here, try to get it from AutorestOptions and fallback to getSecurityInfoFromModel